pages/docs/manual/latest/bind-to-ts-function.mdx

```diff
@@ -0,0 +1,24 @@
+---
+title: "Bind to TS Function"
+description: "Interop with functions between ReScript and TypeScript"
+canonical: "/docs/manual/latest/bind-to-ts-function"
+---
+
+# Bind to TS Function
+
+## Functions and Function Components
+
+ReScript functions are exported as JS functions of the corresponding type.
+So for example a ReScript function `foo : int => int` is exported as a JS function from numbers to numbers.
+
+If named arguments are present in the ReScript type, they are grouped and exported as JS objects. For example `foo : (~x:int, ~y:int) => int` is exported as a JS function from objects of type `{x:number, y:number}` to numbers.
```

Comment:
I'm not sure it's a regression or breaking change, but this isn't true in v11

```res
@genType
let fn1 = (~foo, ~bar) => {
  Js.log2(foo, bar)
}
```

is compiled to

```ts
export const fn1: <T1,T2>(foo:T1, bar:T2) => void = gentypeBS.fn1;
```

This is a necessary change as there will eventually be no runtime mappers, but I personally liked that feature.

It would be nice to see it added back to the compiler core, as that feature is still useful from a js interop perspective as well as gentype.

cc @cristianoc

Reply:
So I actually got **bunch** of breaking changes from my TS/ReS hybrid project.

![image](https://github.com/rescript-association/rescript-lang.org/assets/9696352/44bf3d25-a413-419b-a362-0efc2d270dc8)

This should be mentioned in the migration guide

Reply:
Just made https://github.com/rescript-lang/rescript-compiler/issues/6530 to resolve this cc @cristianoc